hashicorp/dev-portal
# feat: update hcp tutorials sidebar

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksupdate-tutorials-sidebar-hashicorp.vercel.app/hcp/tutorials) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202932567024648) 🎟️
- [Figma Link](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12456%3A147293) 🎨 

## 🗒️ What

This PR adds a custom sidebar for the HCP tutorials landing and collection pages. 

## 📸 Design Screenshots

<img width="732" alt="Screen Shot 2022-09-08 at 1 19 25 PM" src="https://user-images.githubusercontent.com/36613477/189217914-79df2ad5-37dc-4bc3-a522-e4e3347327cd.png">


## 🧪 Testing

- [Visit the preview ](https://dev-portal-git-ksupdate-tutorials-sidebar-hashicorp.vercel.app/hcp/tutorials)
- Validate that the sidebar renders correctly, matching the designs. 
- Click on a collection link,  verify that the sidebar is the same.
- Go to a tutorial, shrink down to a mobile viewport, open up the nav menu. Go up one level and verify that the custom sidebar content renders. 
- Go to other product tutorial pages to smoke-test that they are working the same. 

## 💭 Anything else?

In the future, I think we may want to refactor HCP out of the dynamic route generation that other product tutorials use, because of all the one-offs. However, leaving that large refactor out of this work to get a mvp version going.
